modules/core/src/test/java/org/apache/ignite/internal/causality/IncrementalVersionedValueTest.java

```diff
@@ -341,6 +344,51 @@ void testCompleteMultipleFutures() {
         assertThat(future3, willBe(2));
     }
 
+    /**
+     * Tests that {@link IncrementalVersionedValue#dependingOn(IncrementalVersionedValue)} provides causality between 2 different values.
+     */
+    @RepeatedTest(100)
+    public void testDependingOn() {
+        var vv0 = new IncrementalVersionedValue<>(register, () -> 1);
+
+        var vv1 = new IncrementalVersionedValue<>(dependingOn(vv0), () -> 1);
+
+        int token = 1;
+
+        vv0.update(token, (i, e) -> completedFuture(i + 1));
+
+        vv1.update(token, (i, e) -> completedFuture(i + 1));
+
+        register.moveRevision(token);
+
+        // Will always complete in time.
+        vv1.get(token).join();
+
+        assertTrue(vv0.get(token).isDone());
+    }
+
+    /**
+     * Tests that {@link IncrementalVersionedValue#update(long, BiFunction)} closure is called immediately when underlying value is
+     * accessible, i.e. when there were no other updates.
+     */
+    @Test
+    public void testImmediateUpdate() {
```

Comment:
Should this be tested? Is this property a desired property important for the contract of `IncrementalVersionedValue`, or it's just a coincidence that the current implementation behaves like this?

Reply:
Technically, it's undefined, but I'd rather have this as an explicit contract. It is also the sole reason why "dependingOn" doesn't work the way I expected it to work earlier

Reply:
Done